### ISSUE TITLE
annotation and marker include custom View by using PointAnnotation or SymbolLayer(especially on android) 

### ISSUE THREAD

**Issue by panda919** [react-native-mapbox-gl/maps#274]
I gonna create custom marker view with @react-native-mapbox-gl/maps : 7.0.0-rc3 and react-native: 0.59.9 such as below image
![Screenshot_4](https://user-images.githubusercontent.com/25161426/61752388-08977780-adde-11e9-84fd-26ea43cc7ad3.png)
I implement below code .
```
 <MapboxGL.BackgroundLayer>
            <MapboxGL.PointAnnotation
                        key={uuid()}
                        id={uuid()}
                        coordinate={this.state.coordinates}
                    >
                        <MapboxMarker activeID={this.props.activeID} />
             </MapboxGL.PointAnnotation>
 </MapboxGL.BackgroundLayer>
```
in here ```<MapboxMarker activeID={this.props.activeID} />``` is custom view.
```
<View style={styles.task}>
                <View style={styles.nameBox}>
                    <Text style={styles.name}>
                       {address}
                    </Text>
                </View>
</View>
```
So I get expected result on ios, but I can not see this custom view on android.
Also, I got PointAnnotation is deprecated and SymbolLayer  is recommened in recent version.
But anyway my above code is working on ios,only doesn't work android.
My question.
1. how to fix above code for android?
2. if i have to use SymbolLayer  , how to create custom view marker by using SymbolLayer .

**Comment by mfazekas** [comment]
PointAnnotation is broken on android at the moment, because it requires nontrivial changes for the new mapbox libs. PR's are wellcome.

**Comment by panda919** [comment]
thanks.
then, how to implement custom view marker by using SymbolLayer ?

**Comment by justincrich** [comment]
I'm blocked on this too ... any guidance would be appreciated.

**Comment by mfazekas** [comment]
SymbolLayer is not a complete replacement of PointAnnotation. SymbolLayer is only replacement if you want to render icons. For custom view markers are not supported at the moment on android.

**Comment by AbdulkareemBarghouthi** [comment]
Any solutions found for this issue?

**Comment by jaysquared** [comment]
I render the views as png with react-native-view-shot and use it as SymbolLayer icons. Hacky, but works fine with good performance for a few views.

**Comment by frankrowe** [comment]
PointAnnotation should work with custom views on iOS and Android now

**Comment by ascorbic** [comment]
Is PointAnnotation not deprecated?

**Comment by naftalibeder** [comment]
@frankrowe Can you point to a commit where this was changed? It still seems broken on Android and I'm having trouble finding what you're referring to.

**Comment by ifaouibadi** [comment]
In case you couldn't view your PointAnnotation in the MapView in android just add 
<MapboxGL.PointAnnotation style={{zIndex: 1000}} 
:D

**Comment by lrpinto** [comment]
I am able to render a point annotation in Android with a custom view with no issues, except that the content of the point annotation is not updating after re-rendering / state has been updated.

If anyone has a hack for this one please share :)

**Comment by lrpinto** [comment]
Little update here and short explanation on how to achieve callout with ShapeSource, SymbolLayer and PointAnnotation  https://github.com/react-native-mapbox-gl/maps/issues/532#issuecomment-580347090

As for the rendering issue I was having only on Android device, where the data displayed in the callout would never update after mounting the first one, even tough the render and state update were clearly seen..., I sorted by

1. lifting state to top level parent above the map, in my case that was a screen component and in addition
2. trigger a fake state update between re-rendering the callout, like this

```
setSelectedFeature = selectedFeature => {
        /* Before actually updating the selected feature, I am setting the selectedFeature to 'undefined' on the screen state on purpose. This is meant to force the removal of a previous point annotation from the tree, and make sure the next point annotation for the newly updated feature presents the new data. This was due to a bug only observed in Android, that I could not solve otherwise. */
        this.setState({ selectedFeature: undefined }, () => {
            this.setState({ selectedFeature })
        })
    }


and in render

```
{selectedFeature && <PointAnnotation .... 
```

Marker with Custom Callout in iOS Achievement Unlocked. Good luck all and ty for your help and great work. :) 

Also I have several layers with markers and cluster, different data driven styling and 100s of points, and it is all working functionality well. 

Last thing to solve is zIndex not behaving well in Android, as the callout is not being put on top of all layers. Chasing this one yet...

**Comment by mfazekas** [comment]
MarkerView should work for this.

**Comment by sauhardar** [comment]
I tried to use a PointAnnotation component on Android as a marker but soon realized it is not interactive. Since I need an interactive component (on which has `onPress` functionality, I reverted to using MarkerView. However, I realized that MarkerView has very poor performance on Android. While moving around the map, the component seems to lag. Any replacement component that is also interactive? @mfazekas @lrpinto @jaysquared ?
